File: 242.valid-anagram.cpp
#include <bits/stdc++.h>
using namespace std;
/*
 * @lc app=leetcode id=242 lang=cpp
 *
 * [242] Valid Anagram
 */

// @lc code=start
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.size() != t.size())
        {
            return false;
        }

        unordered_map<char, int> umap1;
        unordered_map<char, int> umap2;

        for (int i = 0; i < s.size(); i++)
        {
            umap1[s[i]]++;
            umap2[t[i]]++;
        }
        

        return umap1 == umap2;
    }
};
// @lc code=end

File: Java/Grind75/Week 1/242.valid-anagram.java
import java.util.*;

/*
 * @lc app=leetcode id=242 lang=java
 *
 * [242] Valid Anagram
 */

// @lc code=start
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        Map<Character, Integer> sMap = new HashMap<>();
        for (Character character : s.toCharArray()) {
            sMap.put(character, sMap.getOrDefault(character, 0) + 1);
        }
        for (Character character : t.toCharArray()) {
            sMap.put(character, sMap.getOrDefault(character, 0) - 1);
            if (sMap.get(character).equals(-1)) {
                return false;
            }
        }
        return true;
    }
}
// @lc code=end
